OsmAnd/src/net/osmand/plus/views/AsynchronousResampler.java

```diff
@@ -0,0 +1,282 @@
+package net.osmand.plus.views;
+
+import android.graphics.Color;
+import android.os.AsyncTask;
+
+import net.osmand.plus.GPXUtilities;
+import net.osmand.plus.views.OsmandMapTileView;
+import net.osmand.util.MapUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AsynchronousResampler extends AsyncTask<String,Integer,String> {
+
+    private OsmandMapTileView view;
+    private Renderable.RenderableSegment rs;
+    private Renderable.RenderType renderType;
+    private double param1, param2;
+
+    private List<GPXUtilities.WptPt> culled = null;
+
+
+    public AsynchronousResampler(Renderable.RenderType rt, OsmandMapTileView view,
+                                 Renderable.RenderableSegment rs, double param1, double param2) {
+        this.view = view;
+        this.rs = rs;
+        this.renderType = rt;
+
+        this.param1 = param1;
+        this.param2 = param2;
+    }
+
+    public int getColor2(double percent) {
+
```

Comment:
It is better to have that piece of code somewhere in Algorithms class with proper explanation what is doing it looks very generic.